### PR TITLE
Mock redis in frontstage tests

### PR DESCRIPTION
**Pull request by matthew-robinson-ons**
# What and why?
The unit tests in frontstage were only able to be run with an instance of redis also running, otherwise multiple tests would fail. This Pr addresses that by introducing a redis mock (fakeredis) to be used in testing. 
# How to test?
Run frontstage unit tests with and without a running instance of redis, all tests should pass both times. 
# Jira
https://jira.ons.gov.uk/secure/RapidBoard.jspa?rapidView=1493&projectKey=RAS&view=detail&selectedIssue=RAS-639

### CODE REVIEW

File: frontstage/__init__.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import fakeredis as fakeredis
```
Comment (insacuri) [inline frontstage/__init__.py:1]: Why the `as fakeredis` at the end?

Comment (matthew-robinson-ons) [reply frontstage/__init__.py:1]: Was autocompleted when i added fakeredis will remove it before merging.

File: frontstage/__init__.py [hunk 2]
```diff
@@ -48,6 +49,10 @@
     frame_options="DENY",
 )
 redis = redis.Redis(host=app.config["REDIS_HOST"], port=app.config["REDIS_PORT"], db=app.config["REDIS_DB"])
+if app.config["TESTING"]:
```
Comment (insacuri) [inline frontstage/__init__.py:52]: This should be written as:
```python
if app.config["TESTING"]:
    <make a fake redis>
else:
    <make a real one>
```

Also, can you put a comment in each path to say `logger.info('Using real redis')` and `logger.info('Using fake redis')` so there's less ways for a mistake to happen

**Review by insacuri — CHANGES_REQUESTED** [review]

**Comment by insacuri** [comment]
Tested it both locally and in k8s and it seems happy.  Ran the acceptance tests and it's good too.
Adding the log line to say if it's using the real or fake one inspires waaay more confidence that a mistake isn't being made.  Also, the fact that the `TESTING` config item is hardcoded as False everywhere except the TestingConfig means it should be incredibly obvious if the wrong config is used.

**Review by insacuri — APPROVED** [review]

**Pull request merged**